### PR TITLE
Document exported API types

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Update docs for https://github.com/grammyjs/grammY/issues/127.

Please only merge once https://github.com/grammyjs/grammY/issues/127 is merged! This should not go live before 1.10.

### CODE REVIEW

File: site/docs/guide/api.md [hunk 1]
```diff
@@ -73,6 +73,53 @@ As an example, some specific properties in some specific methods have to be `JSO
...
+
+If you do not change your `tsconfig.json` file as described above, it may happen that your code editor suggests in autocomplete to import types from `grammy/out/client` or something.
+**All paths starting with `grammy/out` are internal. Do not use them.**
+They could be changed arbitrarily at any point in time, so we strongly advise you to import from `grammy/types` instead.
+
+:::
+
+#### Type Definitions on Deno
+
+On Deno, you can simply import type definitions from `types.ts` instead of `mod.ts`.
```
Comment (roj1512) [inline site/docs/guide/api.md:117]:
```suggestion
On Deno, you can simply import type definitions from `types.ts`:
```

Comment (KnorpelSenf) [reply site/docs/guide/api.md:117]: I agree the phrasing is bad. Can we try to keep the point that the file is located next to `mod.ts`?
```suggestion
On Deno, you can simply import type definitions from `types.ts`, which is right next to `mod.ts`.
```

Comment (roj1512) [reply site/docs/guide/api.md:117]: You can do that, but remove the comma. and make the dot a semi-colon.

Comment (KnorpelSenf) [reply site/docs/guide/api.md:117]:
```suggestion
On Deno, you can simply import type definitions from `types.ts` which is right next to `mod.ts`:
```
?

Comment (roj1512) [reply site/docs/guide/api.md:117]: Exactly

**Review by roj1512 — APPROVED** [review]

**Review by habemuscode — APPROVED** [review]

File: site/docs/guide/api.md [hunk 2]
```diff
@@ -73,6 +73,53 @@ As an example, some specific properties in some specific methods have to be `JSO
...
+
+```ts
+import { type Chat } from "grammy/types";
+```
+
+Officially, Node.js only supports importing from sub-paths properly since Node.js 16.
+Consequently, TypeScript requires the `moduleResolution` to be set to `node16` or `nodenext`.
+Adjust your `tsconfig.json` accordingly and add the highlighted line:
+
+```jsonc{4}
```
Comment (habemuscode) [inline site/docs/guide/api.md:95]: ` ```json{4}` instead of ` ```jsonc{4}`?

**Pull request merged**